File: src/apps/Andriod/app/src/main/java/com/dkellycollins/triggerfinger/managers/events/IEventInterceptor.java
package com.dkellycollins.triggerfinger.managers.events;

/**
 * Provides the ability to intercept events.
 */
public interface IEventInterceptor {

    /**
     * Indicates if the interceptor handles the provided event.
     * @param event The event to test.
     * @return True if the interceptor can handle the event. False otherwise.
     */
    boolean handlesMessage(IEvent event);

    /**
     * Invokes the interceptor with the provided event.
     *
     * @param event The event.
     */
    void invoke(IEvent event);
}

File: src/apps/Android-Kotlin/app/src/main/kotlin/com/dkellycollins/triggerfinger/managers/events/IEventInterceptor.kt
package com.dkellycollins.triggerfinger.managers.events

/**
 * Provides the ability to intercept events.
 */
interface IEventInterceptor {

    /**
     * Indicates if the interceptor handles the provided event.
     * @param event The event to test.
     * *
     * @return True if the interceptor can handle the event. False otherwise.
     */
    fun handlesMessage(event: IEvent): Boolean

    /**
     * Invokes the interceptor with the provided event.

     * @param event The event.
     */
    operator fun invoke(event: IEvent)
}
